### ISSUE TITLE
Endpoints that simulates connection errors

### ISSUE THREAD

**Issue by stefansundin** [kennethreitz/httpbin#420]
Hi there,

I have had the need to test HTTP clients in order to make sure they are robust, in this case I'm looking to add retry logic in case of connection errors. Most libraries throw some kind of exception when these errors happen, and it may be many different exceptions depending on where in the request something went wrong.

In the case of Ruby and its built-in net/http library, some of the errors that I've had to check for are:
- Net::OpenTimeout
- Net::ReadTimeout
- SocketError
- Errno::ECONNREFUSED
- Errno::ECONNRESET
- OpenSSL::SSL::SSLError
- EOFError

Basically, it would be cool to have:
- endpoints that breaks the connection in different ways to raise the exceptions listed above
- an endpoint to pick one of those errors at random
- an endpoint that gives a request a certain chance of succeeding, which would be useful when testing retry logic

Thanks!

**Comment by sigmavirus24** [comment]
This has been requested before (you can find those discussions on the issue tracker). httpbin is not a server for testing every possible behaviour. Instead, it sounds like what you really want is [pathod](http://docs.mitmproxy.org/en/latest/pathod/intro.html).
